Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/KeyboardInputFieldBase.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine.UI;
+#if WINDOWS_UWP
+using UnityEngine;
+using UnityEngine.EventSystems;
+using Microsoft.MixedReality.Toolkit.Input;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.UI
+{
+    /// <summary>
+    /// Base class explicitly launching Windows Mixed Reality's system keyboard for InputField and TMP_InputField
+    /// To be attached to the same GameObject with either of the components
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public abstract class KeyboardInputFieldBase<T> : MixedRealityKeyboardBase
+#if WINDOWS_UWP
+    , IDeselectHandler, IMixedRealityPointerHandler
+#endif
+    where T : Selectable
+    {
+        [Experimental]
+        protected T inputField;
+
+        void OnValidate()
+        {
+            inputField = GetComponent<T>();
+
+            DisableRaycastTarget(Text(inputField));
+            DisableRaycastTarget(PlaceHolder(inputField));
+        }
+
+        private void DisableRaycastTarget(Graphic graphic)
+        {
+            if (graphic != null)
+            {
+                graphic.raycastTarget = false;
+            }
+        }
+
+#if WINDOWS_UWP
+
+        protected virtual void Awake()
+        {
+            if ((inputField = GetComponent<T>()) == null)
+            {
+                Destroy(this);
+            }
+
+            Debug.LogWarning($"There is no {typeof(T).ToString()} on GameObject {name}, removing this component");
+        }
+
+        #region IDeselectHandler implementation
+
+        public void OnDeselect(BaseEventData eventData) => HideKeyboard();
+
+        #endregion
+
+        #region IMixedRealityPointerHandler implementation
+
+        public void OnPointerDown(MixedRealityPointerEventData eventData) { }
```

Comment:
Do these need to be public? Any time you add public fields, it increases API surface. [Add public fields with care](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/Contributing/CodingGuidelines.html#add-new-public-fields-properties-methods-and-serialized-private-fields-with-care)

Reply:
Yes they do because they are the implementation of the IMixedRealityPointerHandler. People can't cast this object to that interface and then call the expected functions if they are not public